### ISSUE TITLE
NNAPI doesn't support tensors with rank 0 - Mobilenet

### ISSUE THREAD

**Issue by ramtin2080** [tensorflow/tensorflow#24940]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Windows 10 for retraining, Linux Ubuntu 16.04 for converting to tflite model
- Mobile device: Sony Xperia Z1 - Android 9 (AICP ROM)
- TensorFlow installed from : binary
- TensorFlow version: b'v1.12.0-rc2-3-ga6d8ffae09' 1.12.0
- Python version: 3.6

**Describe the current behavior**
I retrained "https://tfhub.dev/google/imagenet/mobilenet_v1_050_160/quantops/classification/1" using this script: https://github.com/tensorflow/hub/blob/master/examples/image_retraining/retrain.py.

Then I converted the output to tflite format using the following command:
```
tflite_convert \
  --output_file=graph.tflite \
  --graph_def_file=retrained_graph.pb \
  --inference_type=QUANTIZED_UINT8 \
 --inference_input_type=QUANTIZED_UINT8 \
  --input_arrays=Placeholder \
  --output_arrays=final_result \
  --input_shapes=1,160,160,3 \
  --mean_values=128 \
  --std_dev_values=127 
```
I moved graph.tflite (the output of the above command) to assets folder of this project: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/java/demo

I did all the necessary changes (such as changing the graph name, input size, etc.) in the ImageClassifierQuantizedMobileNet class.

Everything is right as long as CPU is selected from the device list, but as soon as I select NNAPI, the app crashes.

I think the problem is from the pre-trained models that are available on tfhub. Since when I convert the stock mobilenet_v1_1.0_224.pb from "http://download.tensorflow.org/models/mobilenet_v1_2018_02_22/mobilenet_v1_1.0_224.tgz "
the problem is gone.

Log:
```
1975-04-19 08:19:20.212 14862-15282/android.example.com.tflitecamerademo E/tflite: NNAPI doesn't support tensors with rank 0 (index 91 name module_apply_default/hub_input/Mul/y)
1975-04-19 08:19:20.212 14862-15282/android.example.com.tflitecamerademo E/tflite: Returning error since TFLite returned failure nnapi_delegate.cc:742.
1975-04-19 08:19:20.212 14862-15282/android.example.com.tflitecamerademo E/tflite: Failed to build graph for NNAPI
    
    --------- beginning of crash
1975-04-19 08:19:20.217 14862-15282/android.example.com.tflitecamerademo E/AndroidRuntime: FATAL EXCEPTION: CameraBackground
    Process: android.example.com.tflitecamerademo, PID: 14862
    java.lang.IllegalArgumentException: Internal error: Failed to run on the given Interpreter: 
        at org.tensorflow.lite.NativeInterpreterWrapper.run(Native Method)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:140)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:216)
        at org.tensorflow.lite.Interpreter.run(Interpreter.java:195)
        at com.example.android.tflitecamerademo.ImageClassifierQuantizedMobileNet.runInference(ImageClassifierQuantizedMobileNet.java:95)
        at com.example.android.tflitecamerademo.ImageClassifier.classifyFrame(ImageClassifier.java:125)
        at com.example.android.tflitecamerademo.Camera2BasicFragment.classifyFrame(Camera2BasicFragment.java:804)
        at com.example.android.tflitecamerademo.Camera2BasicFragment.access$1200(Camera2BasicFragment.java:78)
        at com.example.android.tflitecamerademo.Camera2BasicFragment$8.run(Camera2BasicFragment.java:699)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
```

**Comment by wosiu** [comment]
related #24676

**Comment by ramtin2080** [comment]
The problem is solved using @impjdi solution in #25131

**Comment by wosiu** [comment]
@ramtin2080 Could you explain shortly what operation caused the issue in your case and how you fixed it?
I read the #25131, however I don't use any normalization (despite BatchNorm) and still get an issue.

**Comment by ramtin2080** [comment]
@wosiu In this case the problem was from module_apply_default/hub_input/Mul/y and module_apply_default/hub_input/Sub/y , which are for normalizing the image. these ops have an empty dimension and NNAPI & GPU delegate currently do not support support dynamic tensor sizes. So in my case, we chop off these ops which cause the failure and do the normalization with Java inside the Android app. 
I think an alternative way is to specify the tensor dimensions for the ops.
